core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -372,9 +372,10 @@ object TopicCommand extends Logging {
                 val partitionDesc = PartitionDescription(
                   topic,
                   partitionId,
-                  leader = if (leaderIsrEpoch.isEmpty) None else Option(leaderIsrEpoch.get.leaderAndIsr.leader),
+                  leader = if (leaderIsrEpoch.isEmpty || !liveBrokers.contains(leaderIsrEpoch.get.leaderAndIsr.leader)) None
+                    else Option(leaderIsrEpoch.get.leaderAndIsr.leader),
                   assignedReplicas,
-                  isr = if (leaderIsrEpoch.isEmpty) Seq.empty[Int] else leaderIsrEpoch.get.leaderAndIsr.isr,
+                  isr = if (leaderIsrEpoch.isEmpty) Seq.empty[Int] else leaderIsrEpoch.get.leaderAndIsr.isr.filter(liveBrokers.contains),
```

Comment:
This is sort of the opposite of what I suggested in the JIRA. It is often still useful to know the ISR information even when the broker is offline. Maybe it would be better for now to list the isr as "not available" when using `--bootstrap-server`?

Reply:
Thanks for the comments. Does it make sense to display a nonempty isr for a partition with no leader actually?
`Topic: foo      Partition: 0    Leader: -1      Replicas: 1,3   Isr: 1`

You said it showed the current ISR correctly. I am very intrigued to know the reason.  The output for `--bootstrap-server`, in my view, reflects the correct state for a partition.

Reply:
The ISR is still valid if the leader is offline. It tells you which replica must be brought back online in order to avoid the need for unclean leader election.